### PR TITLE
Add cert-manager integration doc

### PR DESCRIPTION
**Pull request by howardjohn**
This starts the new "Integrations" page, which will include many other
integrations in the future. For now, I have added cert-manager
integration. We have an existing doc for this at
https://istio.io/docs/tasks/traffic-management/ingress/ingress-certmgr/
which I think we can remove, but I did not do that in this PR.

Note to the reviewer: this is explicitly not a task, as we do not
control cert-manager. Having an explicit set of steps and verifications
requires an exact working directions, which is maintained as
cert-manager evolves, etc. The goal here was to link out to cert-manager
docs wherever possible, and just document the very minimal amount needed
to actually get things integrated with Istio. As you can see from the
doc this is fairly minimal - for the most part, its just referencing a
secret crated by cert-manager.

### CODE REVIEW

**Comment by howardjohn** [comment]
Fixes https://github.com/istio/istio/issues/16665
Fixes https://github.com/istio/istio/issues/18869
Fixes https://github.com/istio/istio/issues/19417
Fixes https://github.com/istio/istio/issues/20514

**Comment by howardjohn** [comment]
/hold

Do not merge until https://github.com/istio/istio/pull/22227 is merged

File: content/en/docs/ops/integrations/_index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: Integrations
+description: Other softwares that Istio can integrate with to provide additional functionality.
+weight: 50
```
Comment (frankbu) [inline content/en/docs/ops/integrations/_index.md]:
```suggestion
weight: 60
```

**Comment by howardjohn** [comment]
dependant PR was merged

**Pull request merged**